bindings/python/requirements/build.txt

```diff
@@ -1,3 +1,3 @@
 setuptools>=47
 cython>=0.29
-pyarrow>=3,<3.1
\ No newline at end of file
+pyarrow>=3
```

Comment:
I suspect we need to update to `>=6,<6.1` because of the way we compile/link against pyarrow.